### PR TITLE
dep: update `sinon` dependency

### PR DESCRIPTION
**Pull request by vince-fugnitto**
#### What it does

Fixes: https://github.com/eclipse-theia/theia/issues/10380

The pull-request updates `sinon` (a `devDependency` used for testing) to a version which does not contain critical security vulnerabilities caused by `js-yaml` and `uglify-js`.

#### How to test

1. the CI should successfully pass - both the build and tests
2. the vulnerability should not longer be reported when performing `yarn audit --level=critical`:

```
$ yarn audit --level=critical
yarn audit v1.22.4
47 vulnerabilities found - Packages audited: 1946
Severity: 15 Low | 26 Moderate | 6 High
Done in 12.84s.
```

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

Signed-off-by: vince-fugnitto <vincent.fugnitto@ericsson.com>

### CODE REVIEW

**Comment by vince-fugnitto** [comment]
The `sinon` dependency is marked as `BSD-3-Clause` which is an approved license. We should probably run the manual tool for dash-licenses to update:

image

File: packages/core/src/browser/keyboard/browser-keyboard-layout-provider.spec.ts [hunk 1]
```diff
@@ -26,11 +30,16 @@ import { WindowService } from '../window/window-service';
...
+
 describe('browser keyboard layout provider', function (): void {
 
     let stubOSX: sinon.SinonStub;
     let stubWindows: sinon.SinonStub;
-    let stubNavigator: sinon.SinonStub;
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+    });
```
Comment (paul-marechal) [inline packages/core/src/browser/keyboard/browser-keyboard-layout-provider.spec.ts:42]: Do we need to cleanup on `after`?

**Comment by marcdumais-work** [comment]
> We should probably run the manual tool for dash-licenses to update:

done: https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1577

**Comment by vince-fugnitto** [comment]
The dependency was approved :+1: 
- https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/1577

**Review by paul-marechal — APPROVED** [review]
Odd that we had to convert the properties into getters for this to work but it's also better to use getters for events, so LGTM!

**Review by marcdumais-work — APPROVED** [review]
LGTM, thanks Vince!

**Pull request merged**